daemon/k8s_watcher.go

```diff
@@ -1632,3 +1675,134 @@ func (d *Daemon) deleteK8sPodV1(pod *v1.Pod) {
 	}
 
 }
+
+func (d *Daemon) updateK8sNodeTunneling(k8sNodeOld, k8sNodeNew *v1.Node) error {
+	logger := log.WithFields(logrus.Fields{
+		"K8sNodeName": k8sNodeNew.ObjectMeta.Name,
+	})
+
+	nodeNew := k8s.ParseNode(k8sNodeNew)
+	// Ignore own node
+	if nodeNew.Name == node.GetName() {
+		return nil
+	}
+
+	getIDs := func(node *node.Node, k8sNode *v1.Node) (*identity.IPIdentityPair, string, error) {
+		if node == nil {
+			return nil, "", nil
+		}
+		hostIP := node.GetNodeIP(false)
+		if ip4 := hostIP.To4(); ip4 == nil {
+			logger.Debugf("Skipping ipcache update, HostIP is not an IPv4 address %s", hostIP)
+			return nil, "", nil
+		}
+
+		ciliumIPStr := k8sNode.GetAnnotations()[annotation.CiliumHostIP]
+		ciliumIP := net.ParseIP(ciliumIPStr)
+		if ciliumIP == nil {
+			return nil, "", fmt.Errorf("no/invalid Cilium-Host IP: %s", ciliumIPStr)
+		}
+
+		id := &identity.IPIdentityPair{
+			IP:     ciliumIP,
+			HostIP: hostIP,
+			ID:     identity.ReservedIdentityHost,
+		}
+		return id, ciliumIPStr, nil
+	}
+
+	identityPairNew, ciliumIPStrNew, err := getIDs(nodeNew, k8sNodeNew)
+	if err != nil {
+		return err
+	}
```

Comment:
There are several possible nil dereference of `identityPairNew` below as `getIDs()` can return nil:
```
+	getIDs := func(node *node.Node, k8sNode *v1.Node) (*identity.IPIdentityPair, string, error) {
+		if node == nil {
+			return nil, "", nil
+		}
+		hostIP := node.GetNodeIP(false)
+		if ip4 := hostIP.To4(); ip4 == nil {
+			logger.Debugf("Skipping ipcache update, HostIP is not an IPv4 address %s", hostIP)
+			return nil, "", nil
+		}
```

I think you want to bail out if `getIDs()` returns nil

Reply:
Done